src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpFormatting.cs

```diff
@@ -0,0 +1,178 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.VisualStudio.IntegrationTest.Utilities;
+using Microsoft.VisualStudio.IntegrationTest.Utilities.Input;
+using Roslyn.Test.Utilities;
+using Roslyn.VisualStudio.IntegrationTests.Extensions.Editor;
+using Xunit;
+
+namespace Roslyn.VisualStudio.IntegrationTests.CSharp
+{
+    [Collection(nameof(SharedIntegrationHostFixture))]
+    public class CSharpFormatting : AbstractEditorTest
+    {
+        protected override string LanguageName => LanguageNames.CSharp;
+
+        public CSharpFormatting(VisualStudioInstanceFactory instanceFactory)
+            : base(instanceFactory, nameof(CSharpFormatting))
+        {
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void AlignOpenBraceWithMethodDeclaration()
+        {
+            SetUpEditor(@"
+$$class C
+{
+    void Main()
+     {
+    }
+}");
+
+            Editor.FormatDocument();
+            this.VerifyTextContains(@"
+class C
+{
+    void Main()
+    {
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void FormatOnSemicolon()
+        {
+            SetUpEditor(@"
+public class C
+{
+    void Foo()
+    {
+        var x =        from a             in       new List<int>()
+    where x % 2 = 0
+                      select x   ;$$
+    }
+}");
+
+            this.SendKeys(VirtualKey.Backspace, ";");
+            this.VerifyTextContains(@"
+public class C
+{
+    void Foo()
+    {
+        var x = from a in new List<int>()
+                where x % 2 = 0
+                select x;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void FormatSelection()
+        {
+            SetUpEditor(@"
+public class C {
+    public void M( ) {$$
+        }
+}");
+
+            this.SelectTextInCurrentDocument("public void M( ) {");
+            Editor.FormatSelection();
+            this.VerifyTextContains(@"
+public class C {
+    public void M()
+    {
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Formatting)]
+        public void PasteCodeWithLambdaBody()
+        {
+            SetUpEditor(@"
+using System;
+class Program
+{
+    static void Main()
+    {
+        Action a = () =>
+        {
+            using (null)
+            {
+                $$
```

Comment:
Decided they were also redundant after talking with @rchande about it. I can look to see if there are unit tests for these as well.